tembo-operator/src/dedicated_networking.rs

```diff
@@ -0,0 +1,568 @@
+    } else {
+        format!("{}-dedicated", cdb_name)
+    };
+
+    let lb_scheme = if is_public {
+        "internet-facing"
+    } else {
+        "internal"
+    };
+    let lb_internal = if is_public { "false" } else { "true" };
+
+    info!(
+        "Applying Service: {} in namespace: {} with type: {} and scheme: {}",
+        service_name, namespace, service_type, lb_scheme
+    );
+
+    let mut annotations = serde_json::Map::new();
+    annotations.insert(
+        "external-dns.alpha.kubernetes.io/hostname".to_string(),
+        serde_json::Value::String(format!("{}.{}", service_name, "example.com")),
+    );
+
+    if service_type == "LoadBalancer" {
+        annotations.extend([
+            (
+                "service.beta.kubernetes.io/aws-load-balancer-internal".to_string(),
+                serde_json::Value::String(lb_scheme.to_string()),
+            ),
+            (
+                "service.beta.kubernetes.io/aws-load-balancer-scheme".to_string(),
+                serde_json::Value::String(lb_internal.to_string()),
+            ),
+            (
+                "service.beta.kubernetes.io/aws-load-balancer-nlb-target-type".to_string(),
+                serde_json::Value::String("ip".to_string()),
+            ),
+            (
+                "service.beta.kubernetes.io/aws-load-balancer-type".to_string(),
+                serde_json::Value::String("nlb-ip".to_string()),
+            ),
+            (
+                "service.beta.kubernetes.io/aws-load-balancer-healthcheck-protocol".to_string(),
+                serde_json::Value::String("TCP".to_string()),
+            ),
+            (
+                "service.beta.kubernetes.io/aws-load-balancer-healthcheck-port".to_string(),
+                serde_json::Value::String("5432".to_string()),
+            ),
+        ]);
+    }
+
+    let mut labels = serde_json::Map::new();
+    labels.insert(
+        "cnpg.io/cluster".to_string(),
+        serde_json::Value::String(cdb_name.to_string()),
+    );
+    if is_public {
+        labels.insert(
+            "public".to_string(),
+            serde_json::Value::String("true".to_string()),
+        );
+    }
+
+    let mut service_spec = serde_json::Map::new();
+    service_spec.insert(
+        "ports".to_string(),
+        json!([{
+            "name": "postgres",
+            "port": 5432,
+            "protocol": "TCP",
+            "targetPort": 5432
+        }]),
+    );
+    service_spec.insert(
+        "selector".to_string(),
+        json!({
+            "cnpg.io/cluster": cdb_name,
+            "role": if is_standby { "replica" } else { "primary" }
+        }),
+    );
+    service_spec.insert("sessionAffinity".to_string(), json!("None"));
+    service_spec.insert("type".to_string(), json!(service_type));
+
+    if service_type == "LoadBalancer" {
+        let load_balancer_source_ranges = match cdb.spec.ip_allow_list.clone() {
+            None => vec!["0.0.0.0/0".to_string()],
```

Comment:
instead of defaulting to all IPs, I think we don't add loadBalancerSourceRanges to the spec

Reply:
then when we QA, we need to check it works with or without IP allow list set